### ISSUE TITLE
Converting from string to number with the operator +

### ISSUE THREAD

**Issue by Incrementis** [Incrementis/Javascript-overloaded-plus-operator-#1]
## This copy paste from the wiki should explain the issue - Thank you for reading!
With the help of an example this will be now explained:

_**1.**_ A text box in HTML is generated as follows

`<input type = "text" id = "txt_one">`

_**2.**_ The value from the text box is fetched and converted as follows

`+document.getElementById("bit").value;`

There is an issue here which is the lack of knowledge from my side, how the converting property is working!
If you at any rate know some explanation or better a valid source, please post it under the section "**issues**" in this github project!Thank you very much!

**Comment by dotnetCarpenter** [comment]
`document.getElementById("bit")` get an [Element][Element] that has an id of `bit` and **not** `txt_one`.

So the complete example should be:

```html
<input type = "text" id = "txt_one">
<script>
  let value1 = +document.getElementById("txt_one").value;
</script>
```
Now what data type is `value1`? In this example, it can only be [Number][Number] or [String][String].

`<input type = "text">` is represented in [DOM][DOM] as a [HTMLInputElement][HTMLInputElement] (it inherit from [Element][Element]), which has a `.value` property that returns a String value.

> `value`: string: Returns / Sets the current value of the control.

[The `+` operator, when used in front of a variable or expression][Unary_plus], attempts to convert  it's operand into a number. What is happening here, is explicit [type coercion or _type conversion_][Type coercion]. JavaScript always tried to convert data types into types that can be used where they are evaluated.

For example:
```js
3 + "3" // 33 Number + String = String
3 + +"3" // 6 Number + Number(String) = Number
if (1) { console.log("true") } // 1 -> Boolean(1) -> true
if (0) { console.log("true") } // 0 -> Boolean(0) -> false
```
A simpler answer is available on [stack overflow](https://stackoverflow.com/questions/6682997/what-is-the-purpose-of-a-plus-symbol-before-a-variable).

[Element]:https://developer.mozilla.org/en-US/docs/Web/API/element
[Number]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures#Number_type
[String]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures#String_type
[DOM]: https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model
[HTMLInputElement]: https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement
[Unary_plus]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Arithmetic_Operators#Unary_plus
[Type coercion]: https://developer.mozilla.org/en-US/docs/Glossary/Type_coercion

**Comment by Incrementis** [comment]
"_document.getElementById("bit") get an Element that has an id of bit and **not** txt_one._"

This example has been fixed in the wiki, but was posted wrongly here by copy and paste errors.
Thanks for the hint. It was actually:

1. A text box in HTML is generated as follows

`<input type = "text" id = "bit" value = "2">`

2. The value from the text box is fetched and converted as follows

`+document.getElementById('bit').value;`

Nevertheless your example seems more telling, so I changed it in the wiki to:
_**1.**_ A text box in HTML is generated as follows

```html
<input type="text" id="txt_one">
```

_**2.**_ The value from the text box is fetched and converted as follows

```html
<script>
  var myValue = +document.getElementById("txt_one").value;
</script>
```

Thank you for your valuable criticism, which helped me to understand the logic behind the operator +!
I put your answer in the wiki for reference :).
